src/SqlPersistence/Config/ConnectionManager/ConnectionManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Data.Common;
+using System.Threading.Tasks;
+
+abstract class ConnectionManager
+{
+    public abstract Task<DbConnection> OpenNonContextualConnection();
+    public abstract DbConnection BuildNonContextual();
+    public abstract Task<DbConnection> OpenConnection(IReadOnlyDictionary<string, string> messageHeaders);
+
+    protected async Task<DbConnection> OpenConnection(DbConnection connection)
+    {
+        try
+        {
+            await connection.OpenAsync().ConfigureAwait(false);
+            return connection;
+        }
+        catch
+        {
+            connection.Dispose();
```

Comment:
```suggestion
            connection?.Dispose();
```

Otherwise if connection is null the catch will throw